File: easy/744. NextGreatestLetter/nextGreatest.js
var nextGreatestLetter = function (letters, target) {
    for (const char of letters) {
        if (char.localeCompare(target) > 0) return char;
    }

    return letters[0];
};

console.log(nextGreatestLetter(["c", "f", "j"], "d"));

File: easy/744. NextGreatestLetter/next_greatest.py
from typing import List

def nextGreatestLetter(letters: List[str], target: str) -> str:
    for char in letters:
        if char >target:
            return char
    
    return letters[0]

print(nextGreatestLetter(["c","f","j"], "c"))
print(nextGreatestLetter(["c","f","j"], "a"))
print(nextGreatestLetter(["c","f","j"], "d"))